Signal/src/ViewControllers/ViewControllerUtils.m

```diff
@@ -86,6 +86,44 @@ + (void)setAudioIgnoresHardwareMuteSwitch:(BOOL)shouldIgnore
     }
 }
 
++ (NSString *)examplePhoneNumberForCountryCode:(NSString *)countryCode callingCode:(NSString *)callingCode
+{
+    OWSAssert(countryCode.length > 0);
+    OWSAssert(callingCode.length > 0);
+
+    NSString *examplePhoneNumber = [PhoneNumberUtil examplePhoneNumberForCountryCode:countryCode];
+    OWSAssert(!examplePhoneNumber || [examplePhoneNumber hasPrefix:callingCode]);
+    if (examplePhoneNumber && [examplePhoneNumber hasPrefix:callingCode]) {
+        NSString *formattedPhoneNumber =
+            [PhoneNumber bestEffortFormatPartialUserSpecifiedTextToLookLikeAPhoneNumber:examplePhoneNumber
+                                                         withSpecifiedCountryCodeString:countryCode];
+        if (formattedPhoneNumber.length > 0) {
+            examplePhoneNumber = formattedPhoneNumber;
+        }
+
+        return [NSString
+            stringWithFormat:
+                NSLocalizedString(@"PHONE_NUMBER_EXAMPLE_FORMAT",
+                    @"A format for a label showing an example phone number. Embeds {{the example phone number}}."),
+            [examplePhoneNumber substringFromIndex:callingCode.length]];
+    } else {
+        return @"";
+    }
+}
+
+#pragma mark - Formatting
+
++ (NSString *)formatInt:(int)value
```

Comment:
printf-style formatting (e.g. %d) is not localized.  There's still probably a ton of places I haven't fixed yet.